applications/plugins/SofaCarving/CarvingManager.cpp

```diff
@@ -111,8 +91,7 @@ void CarvingManager::init()
         m_carvingReady = false; 
     }
 
-    if (m_intersectionMethod == nullptr) { msg_error() << "m_intersectionMethod not found. Add an Intersection method in your scene."; m_carvingReady = false; }
-    if (m_detectionNP == nullptr) { msg_error() << "NarrowPhaseDetection not found. Add a NarrowPhaseDetection method in your scene."; m_carvingReady = false; }
+    if (l_detectionNP.get()) { msg_error() << "NarrowPhaseDetection not found. Add a NarrowPhaseDetection method in your scene."; m_carvingReady = false; }
```

Comment:
Don't you search for it if not found? That was the previous behavior. It will break the behavior of the component.

Reply:
You could also set the component state to invalid

Reply:
Done